source/slang/slang-ir.cpp

```diff
@@ -5794,7 +5898,42 @@ namespace Slang
         addInst(i);
         return i;
     }
-
+    IRSPIRVAsmOperand* IRBuilder::emitSPIRVAsmOperandRayPayloadFromLocation(IRInst* inst)
+    {
+        SLANG_ASSERT(as<IRSPIRVAsm>(m_insertLoc.getParent()));
+        const auto i = createInst<IRSPIRVAsmOperand>(
```

Comment:
It seems that you shouldn't have const here because your return type doesn't have const and "i" will be returned.

Reply:
const is used since other functions which do the same thing also set a const ptr for return; likely const is used since in the function we are signaling that the ptr won't change value between instantiation and returning

Reply:
"const" is not a "signal"-ing keyword, and it cannot automatically become non-const.
Something doesn't look right on this code to me.

Reply:
I did some checking, in godbolt -- it should not compile in MSVC (without a flag), nor does it likely emit different ASM...

Reply:
nope, just a cause of warnings; I suppose I will remove all of the const'ness from these functions and if there is a question of why that will be a different discussion